mart/attack/adversary.py

```diff
@@ -132,8 +133,18 @@ def done(self) -> bool:
 
         return False
 
-    def on_run_start(self, adversary, input, target, model, **kwargs):
-        super().on_run_start(adversary, input, target, model, **kwargs)
+    def on_run_start(
+        self,
+        *,
+        adversary: Optional[torch.nn.Module],
+        input: Optional[Union[torch.Tensor, tuple]],
+        target: Optional[Union[torch.Tensor, Dict[str, Any], tuple]],
+        model: Optional[torch.nn.Module],
```

Comment:
If they're truly optional, then you need to provide a default for keyword arguments. My sense is, thought, you need to remove the `Optional` type.

Reply:
Removed.